taas/__init__.py

```diff
@@ -141,15 +144,36 @@ def normalise_sheet(raw, mapping):
 
         Raw is the data as produced from read_sheet_csv().
         Mapping is a dict of cooked->raw pairs (eg: 'label' : 'Preferred Term')
+
+        Fields containing a . in their name will be turned into maps. Eg:
+        `label.en`, `label.es` and `label.fr` become three keys in the `label` map.
+        Only one level of embedding is supported.
     """
 
+    fieldmap = make_map(mapping)
+
     cooked = []
 
     for row in raw:
-        item = {}
-        for field in mapping:
-            item[field] = row[mapping[field]]
-        cooked.append(item)
+        cooked_row = {}
+        for label, processor in fieldmap.items():
+            result = processor.emit(row)
+
+            if '.' in label:
+                # TODO: Would be nice to support any number of fields
+                # TODO: Refactor this into a separate, testable function.
+                keys = label.split('.', 1)
+                topkey = keys[0]
+                subkey = keys[1]
+
+                if topkey not in cooked_row:
+                    cooked_row[topkey] = {}
```

Comment:
Just FYI the normal Python way to do autovivification is using `collections.defaultdict` on itself like so ...
```
def make_autovivifying_dict(): return defaultdict(make_autovivifying_dict)
cooked_row = make_autovivifying_dict()
cooked_row['foo']['bar']['baz'] = 'qux'
```
That gets you part of the way, but what you want is a nested dictionary, probably something along these lines would suit:
http://stackoverflow.com/questions/14692690/access-nested-dictionary-items-via-a-list-of-keys